packages/database/src/core/view/Change.ts

```diff
@@ -17,59 +17,68 @@
 
 import { Node } from '../snap/Node';
 
-/**
- * @param type The event type
- * @param snapshotNode The data
- * @param childName The name for this child, if it's a child event
- * @param oldSnap Used for intermediate processing of child changed events
- * @param prevName The name for the previous child, if applicable
- */
-export class Change {
-  constructor(
-    public type: string,
-    public snapshotNode: Node,
-    public childName?: string,
-    public oldSnap?: Node,
-    public prevName?: string | null
-  ) {}
+/** Event type for a child added */
+export const CHANGE_TYPE_CHILD_ADDED = 'child_added';
 
-  static valueChange(snapshot: Node): Change {
-    return new Change(Change.VALUE, snapshot);
-  }
+/** Event type for a child removed */
+export const CHANGE_TYPE_CHILD_REMOVED = 'child_removed';
 
-  static childAddedChange(childKey: string, snapshot: Node): Change {
-    return new Change(Change.CHILD_ADDED, snapshot, childKey);
-  }
+/** Event type for a child changed */
+export const CHANGE_TYPE_CHILD_CHANGED = 'child_changed';
 
-  static childRemovedChange(childKey: string, snapshot: Node): Change {
-    return new Change(Change.CHILD_REMOVED, snapshot, childKey);
-  }
+/** Event type for a child moved */
+export const CHANGE_TYPE_CHILD_MOVED = 'child_moved';
 
-  static childChangedChange(
-    childKey: string,
-    newSnapshot: Node,
-    oldSnapshot: Node
-  ): Change {
-    return new Change(Change.CHILD_CHANGED, newSnapshot, childKey, oldSnapshot);
-  }
+/** Event type for a value change */
+export const CHANGE_TYPE_VALUE = 'value';
```

Comment:
Can these `const` variables be `const enum`? Can shave a little bit of size.

Reply:
Also will be a bit easier to use.

Reply:
Done. Thanks for the suggestion

Reply:
-500 bytes by the way